chia/types/weight_proof.py

```diff
@@ -100,3 +103,63 @@ class WeightProof(Streamable):
     sub_epochs: List[SubEpochData]
     sub_epoch_segments: List[SubEpochChallengeSegment]  # sampled sub epoch
     recent_chain_data: List[HeaderBlock]
+
+
+@streamable
+@dataclass(frozen=True)
+class SubSlotDataV2(Streamable):
```

Comment:
1. If it's an empty subslot everything will be None right?
2. If we combine the Optionals into one structure, we can save some space (one byte per Optional) and maybe make it less complex

Reply:
Basically for some of these objects, if one of them is present, then the rest will also be present. So instead of having an Optional for each, you can group them up into another stremable, and just have one big Optional[OtherStreamable]

Reply:
i dont think that simplify things because of edge cases and the fact that we replace the proofs for None in challenge slots that are not sampled, we would still need to have optionals inside the different structures